### DISCUSSION TITLE
Q: Can I verify RISC Zero proofs on chains other than Ethereum?

### DISCUSSION THREAD

**Discussion by pdg744** [risc0/risc0#1526] [Q&A] [answered]
A common question we hear:
*Can I verify RISC Zero proofs on chains other than Ethereum?*

**Comment by pdg744** [comment, marked as answer]
A: In order to integrate RISC Zero's proofs with on-chain applications, you just need to post proofs on-chain and verify them with an on-chain verifier. Docs for the verifier contract are [here](https://dev.risczero.com/api/blockchain-integration/contracts/verifier)

Currently, the only chain where we have deployed a verifier is Sepolia. Deploying a verifier on other EVM chains should be straight-forward. For non-EVM chains, you may need to build your own on-chain verifier. 

**Note: This project is not production ready!**

**Reply by ttdung** [reply]
I re-deploy the verifier contracts on Sepolia, and it verifies the proof well. However, when i deploy them on Ethermint, the verifier failed to verify the same proof. Do you have any experience on this?

**Reply by ttdung** [reply]
i make wrong input encoding, it works well now.
